File: src/main/java/cc/duduhuo/git/conflict/BuiltInColor.java
package cc.duduhuo.git.conflict;

import cc.duduhuo.git.conflict.model.MarkColor;

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/25 21:28
 * Description:
 * Remarks:
 * =======================================================
 */
public interface BuiltInColor {
    String AUTO_SCHEME_NAME = "Auto";
    String INTELLIJ_SCHEME_NAME = "IntelliJ";
    String DARCULA_SCHEME_NAME = "Darcula";

    String DEFAULT_SCHEME = AUTO_SCHEME_NAME;

    MarkColor AUTO = new MarkColor(AUTO_SCHEME_NAME, true, 0x9fe3d6, 0xd9f4ef, 0x9fd2ff, 0xd9edff,
        0x2f7366, 0x25403b, 0x2f628f, 0x25394b);
    MarkColor INTELLIJ = new MarkColor(INTELLIJ_SCHEME_NAME, true, 0x9fe3d6, 0xd9f4ef, 0x9fd2ff, 0xd9edff);
    MarkColor DARCULA = new MarkColor(DARCULA_SCHEME_NAME, true, 0x2f7366, 0x25403b, 0x2f628f, 0x25394b);
}

File: src/main/java/cc/duduhuo/git/conflict/BuiltInColor.kt
package cc.duduhuo.git.conflict

import cc.duduhuo.git.conflict.model.MarkColor
import cc.duduhuo.git.conflict.tool.ext.transparent
import java.awt.Color

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/25 21:28
 * Description:
 * Remarks:
 * =======================================================
 */
@Suppress("UseJBColor")
object BuiltInColor {
    const val DEFAULT_SCHEME_NAME = "VS Code"

    private const val HEADER_TRANSPARENCY = 0.5f
    private const val CONTENT_TRANSPARENCY = 0.4f
    private const val RULER_TRANSPARENCY = 1f

    private val currentBaseColor = Color(0x40C8AE).transparent(HEADER_TRANSPARENCY)
    private val incomingBaseColor = Color(0x40A6FF).transparent(HEADER_TRANSPARENCY)
    private val commonBaseColor = Color(0x606060).transparent(0.4f)

    val DEFAULT = MarkColor(
        DEFAULT_SCHEME_NAME, true,
        currentBaseColor,
        currentBaseColor.transparent(CONTENT_TRANSPARENCY),
        incomingBaseColor,
        incomingBaseColor.transparent(CONTENT_TRANSPARENCY),
        commonBaseColor,
        commonBaseColor.transparent(CONTENT_TRANSPARENCY)
    )
}
